### PR TITLE
gcc: align with linux version

### PR DESCRIPTION
**Pull request by iMichka**
- add missing flags
- install gfortran symlinks
- add postinstall step

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/gcc.rb [hunk 1]
```diff
@@ -120,16 +120,42 @@ def install
...
+      end
 
       bin.install_symlink bin/"gfortran-#{version_suffix}" => "gfortran"
+
+      on_linux do
+        # Only the newest brewed gcc should install gfortan libs as we can only have one.
+        lib.install_symlink lib/"gcc/#{version_suffix}/libgfortran.so"
+        lib.install_symlink lib/"gcc/#{version_suffix}/libgfortran.a"
+        lib.install_symlink lib/"gcc/#{version_suffix}/libgfortran.so.5"
+      end
```
Comment (carlocab) [inline Formula/gcc.rb:158]: Is there any reason we don't do this on macOS too?

Comment (carlocab) [reply Formula/gcc.rb:158]: Also, any reason not to do this:
```suggestion
      on_linux do
        # Only the newest brewed gcc should install gfortan libs as we can only have one.
        lib.install_symlink Dir[lib/"gcc/#{version_suffix}/libgfortran.*"]
      end
```

Comment (iMichka) [reply Formula/gcc.rb]: I changed the syntax to be more generic as asked above.

To the "why do you need to do this on Linux": if I don't the linker will fail to find the libraries as they are in the `gcc/10.2.0_4/lib/gcc/10` subfolder. But I don't really know why this is not needed on macOS. Maybe someone from @Homebrew/linux  might have an idea.

**Comment by iMichka** [comment]
This is open since May 3 and there was no strong review against these changes, so I am going to merge this to move forward with the migration.

**Pull request merged**